File: problems/s/securedoors/securedoors.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n = int(input())

    log = list()
    for _ in range(n):
        command, name = input().split()
        if command == "entry":
            if name in log:
                print(f"{name} entered (ANOMALY)")
            else:
                log.append(name)
                print(f"{name} entered")
            
        else:
            if name not in log:
                print(f"{name} exited (ANOMALY)")
            else:
                log.remove(name)
                print(f"{name} exited")


if __name__ == '__main__':
    main()

File: problems/s/securedoors/securedoors.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;

    cin >> n;
    cin.ignore();

    set<string> log;
    for (int i=0; i<n; i++) {
        string command, name;

        cin >> command >> name;
        cin.ignore();

        if (command == "entry") {
            if (log.find(name) != log.end()) {
                cout << name << " entered (ANOMALY)" << endl;
            }
            else {
                log.insert(name);
                cout << name << " entered" << endl;
            }
        }
        else {
            if (log.find(name) == log.end()) {
                cout << name << " exited (ANOMALY)" << endl;
            }
            else {
                log.erase(name);
                cout << name << " exited" << endl;
            }
        }
    }
}